### PR TITLE
Hide Downloads section under Toggle

### LINKED ISSUE #43807: Hide Downloads section under Toggle
**Design**
AkVNGImLgSqCObTQ3idVn7-fi-14340%3A190958

**P2 post**
peCask-Dc-p2 

**Acceptance criteria**

- [x] For new stores, or existing stores creating new products, the Download section should be hidden by default under an `off` toggle, with the label `Include downloads`
- [x] For existing stores with existing products, if the Download section has no files, then the Download section should be hidden behind an `off` toggle
- [x] For existing stores with existing products, if the Download section has files already uploaded, then the Download section should be shown under an `on` toggle
- [x] When the user toggles on `Include downloads`, the Download section should appear below
- [x] If the user uploads a file, then toggles off `Include downloads`, then Saves or Updates the product, then toggles on `Include downloads`, the previously uploaded file should re-appear. (This aligns with how it currently works in our legacy editor, gif below.)

Screen Capture on 2024-01-19 at 15-03-33

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #43807

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-virtual-downloadable` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product`
4. Under `General` tab and the bottom of the page a new `Downloads` section should be shown only when the `Include downloads` checkbox is checked.
5. The `Include downloads` checkbox should be unchecked by default.
6. When loading the page, if the `Download` section has no files, then the `Include downloads` checkbox should be unchecked.
7. When loading the page, if the `Download` section has files, then the `Include downloads` checkbox should be checked.
8. Any change to the download list should remains when saving the product and refreshing the page for the given product.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/blocks/generic/section/editor.scss [hunk 1]
```diff
@@ -74,9 +76,15 @@
 		 * Core removes the "loading" animation styles
 		 * from the button when it's disabled, so we
 		 * let's add them back.
-		 */ 
+		 */
 		.components-button.is-tertiary.is-busy:disabled {
-			background-image: linear-gradient( -45deg, #fafafa 33%, #e0e0e0 0, #e0e0e0 70%, #fafafa 0 );
+			background-image: linear-gradient(
```
Comment (mdperez86) [inline packages/js/product-editor/src/blocks/generic/section/editor.scss:81]: This was auto-format when saving. Really not part of this PR.

**Review by nathanss — APPROVED** [review]
Working as expected.

I couldn't figure out why it works, but unchecking the checkbox when there are downloads and saving seems to remove the downloads. I guess this is expected. Did you account for this scenario?

Pre-approving, feel free to merge if it's expected.

**Comment by mdperez86** [comment]
> Working as expected.
> 
> I couldn't figure out why it works, but unchecking the checkbox when there are downloads and saving seems to remove the downloads. I guess this is expected. Did you account for this scenario?
> 
> Pre-approving, feel free to merge if it's expected.

That is what it was implemented server side (woo rest api), but for some reason differs from the classic editor. I'll check it out. Nice catch! I will tackle this in a different PR.

**Comment by mdperez86** [comment]
> > Working as expected.
> > I couldn't figure out why it works, but unchecking the checkbox when there are downloads and saving seems to remove the downloads. I guess this is expected. Did you account for this scenario?
> > Pre-approving, feel free to merge if it's expected.
> 
> That is what it was implemented server side (woo rest api), but for some reason differs from the classic editor. I'll check it out. Nice catch! I will tackle this in a different PR.

@nathanss this is the https://github.com/woocommerce/woocommerce/pull/44008 about that.
